lib/bloc/nostr/nostr_bloc.dart

```diff
@@ -0,0 +1,174 @@
+  }
+
+  Future<void> _handleSignEvent(SignEvent action) async {
+    //  to sign the event
+    Map<String, dynamic> signedEvent =
+        await _signEvent(action.eventObject, action.privateKey);
+
+    _eventController.add(signedEvent);
+    action.resolve(signedEvent);
+  }
+
+  Future<void> _handleGetRelays(GetRelays action) async {
+    //  to get the relays
+    List<String> relays = await _fetchRelays();
+    // Perform any processing on the relays if needed
+    action.resolve(relays);
+  }
+
+  Future<void> _handleNip04Encrypt(Nip04Encrypt action) async {
+    // to encrypt the data`
+    String encryptedData = await _encryptData(action.data, action.publicKey);
+    action.resolve(encryptedData);
+  }
+
+  Future<void> _handleNip04Decrypt(Nip04Decrypt action) async {
+    // to decrypt the data
+    String decryptedData =
+        await _decryptData(action.encryptedData, action.privateKey);
+    action.resolve(decryptedData);
+  }
+
+  // Methods to simulate the actual logic
+
+  Future<String> _fetchPublicKey() async {
+    if (nostrPublicKey == null) {
+      // check if key pair already exists otherwise generate it
+      String nostrKeyPair;
+
+      nostrKeyPair = await _breezLib.getNostrKeyPair().catchError((error) {
+        throw error.toString();
+      });
+
+      int index = nostrKeyPair.indexOf('_');
+      nostrPrivateKey = nostrKeyPair.substring(0, index);
+      nostrPublicKey = nostrKeyPair.substring(index + 1);
+
+      // Write value
+      await _secureStorage.write(
+          key: 'nostrPrivateKey', value: nostrPrivateKey);
+      await _secureStorage.write(key: 'nostrPublicKey', value: nostrPublicKey);
+    }
+
+    return nostrPublicKey;
+  }
+
+  Future<Map<String, dynamic>> _signEvent(
+      Map<String, dynamic> eventObject, String nostrPrivateKey) async {
+    final eventApi = EventApi();
+
+    if (eventObject['pubkey'] == null) {
+      eventObject['pubkey'] = nostrPublicKey;
+    }
+
+    List<dynamic> dynamicList = eventObject['tags'];
+    List<List<String>> stringList =
+        dynamicList.map((innerList) => List<String>.from(innerList)).toList();
+
+    final event = Event(
+      content: eventObject['content'],
+      created_at: eventObject['created_at'],
+      kind: eventObject['kind'],
+      pubkey: eventObject['pubkey'],
+      tags: stringList,
+    );
+
+    if (eventObject['id'] == null || eventObject['id'] == '') {
+      event.id = eventApi.getEventHash(event);
+      eventObject['id'] = event.id;
+    } else {
+      event.id = eventObject['id'];
+    }
+
+    event.sig = eventApi.signEvent(event, nostrPrivateKey);
+
+    if (eventApi.verifySignature(event)) {
+      eventObject['sig'] = event.sig;
+    }
+
+    return eventObject;
+  }
+
+  Future<List<String>> _fetchRelays() async {
+    return ['Relay1', 'Relay2', 'Relay3'];
+  }
+
+  Future<String> _encryptData(String data, String publicKey) async {
+    // Simulating an encryption operation
+    return 'EncryptedData';
```

Comment:
Is this mean we don't support encrypt/decrypt? In that case isn't it better to throw error?

Reply:
I've made the required changes in the new commit.